app/Http/Controllers/User/UsersController.php

```diff
@@ -259,17 +259,40 @@ private function loginWithEmail($email, $password)
         return $user;
     }
 
-    private function loginWithSocialProvider($provider_id, $provider_user_id)
+    private function loginWithSocialProvider($provider_id, $provider_user_id, $request)
     {
         $account = Account::where('provider_id', $provider_id)
             ->where('provider_user_id', $provider_user_id)->first();
-        if (!$account) {
-            return null;
+        if ($account) {
+            return User::with('accounts', 'profile')->whereId($account->user_id)->first();
+        }
+        // Verify a user with the email exist
+        $user = User::with('accounts', 'profile')->where('email', $request->email)->first();
+        // Create user if not exists
+        if (!$user) {
+            $user = User::create([
+                'name'          => $request->name,
+                'first_name'    => $request->first_name,
+                'last_name'     => $request->last_name,
+                'email'         => $request->email,
+                'activated'     => 0,
+                'password'      => \Hash::make(Str::random(10)),
+            ]);
+
+            Profile::create([
+                'user_id' => $user->id
+            ]);
         }
 
-        $user = User::with('accounts', 'profile')->whereId($account->user_id)->first();
+        // Link the social account provider
+        $user->accounts()->create([
+            'project_id' => $request->project_id,
+            'provider_id'      => $provider_id,
+            'provider_user_id' => $provider_user_id,
+        ]);
 
-        return $user;
+
+        return User::with('accounts', 'profile')->whereId($user->id)->first();
     }
 
     /**
```

Comment:
Is there a reason for this line?

Reply:
Yes, we need the response with the updated accounts and profile information

Reply:
Also removed the empty line :)